### PR TITLE
Address Snyk issues

### PR DESCRIPTION
**Pull request by juniortaeza**
### Desired Outcome

This pull request addresses the Snyk issues outlined in this repository.

### Implemented Changes

- upgrades centOS base image to 8
- upgrades ubuntu base image to 23.04

#### Changelog

- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [X] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [X] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [X] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [X] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: tests/test_app_centos/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,8 @@
-FROM centos:7
+FROM centos
+
+RUN cd /etc/yum.repos.d/
+RUN sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
+RUN sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
 
 # Install Python so Ansible can run against node
 RUN yum update -y && yum install -y python3
```
Comment (juniortaeza) [inline tests/test_app_centos/Dockerfile:8]: The following error occurs when upgrading centos:7 to centos:8 

`Error: Failed to download metadata for repo ‘appstream’: Cannot prepare internal mirrorlist: No URLs in mirrorlist`

According to documents, 

`After Dec 31st, 2021, if you need to update your CentOS, you need to change the mirrors to [vault.centos.org](https://vault.centos.org/) where they will be archived permanently.`

**Thus, a proper upgrade to centos:8 is reflected upon the changes in this file.**

More information regarding this issue can be found [here](https://yegorshytikov.medium.com/error-failed-to-download-metadata-for-repo-appstream-cannot-prepare-internal-mirrorlist-no-959768e5f8e5#:~:text=Fix%20Failed%20to%20download%20metadata%20for%20repo&text=that's%20means%20CentOS%208%20will,they%20will%20be%20archived%20permanently.&text=Go%20to%20the%20%2Fetc%2Fyum,repos.).

File: tests/docker-compose.yml [hunk 2]
```diff
@@ -60,4 +60,4 @@ services:
         TERM: xterm
     depends_on:
       - conjur
-      - conjur_cli
+      - conjur_cli
```
Comment (gl-johnson) [inline tests/docker-compose.yml]: I don't think this needed. Also you may as well add a newline to the end here and in tests.sh to make GH happy

Comment (juniortaeza) [reply tests/docker-compose.yml]: Done!

File: tests/docker-compose.yml [hunk 3]
```diff
@@ -60,4 +60,3 @@ services:
         TERM: xterm
     depends_on:
       - conjur
-      - conjur_cli
```
Comment (gl-johnson) [inline tests/docker-compose.yml]: Woops I misread the diff in my previous comment. You can leave the conjur_cli dependency and just address the missing newlines. There's still one in `tests/test.sh`

**Review by jtuttle — APPROVED** [review]

**Pull request merged**